Refactor code to pass references instead of values for input vectors …

spartajet opened the pull request:
…in `splrep` and related functions. Fixes #1234.

spartajet committed:
Refactor code to pass references instead of values for input vectors in `splrep` and related functions. Fixes #1234.

hochej commented:
Thanks for the PR. I agree that it is not good that vectors are passed here instead of their references. However, I think it should be slices `&[f64]` instead of references to vectors `&Vec<f64>`, don't you? 
As this is a breaking change, the version would also have to be adapted. 

@einseler can you also have a look at this?

spartajet commented:
Yes, &[f64] is better, I will fixed it next week. Thanks!

spartajet committed:
fix(core): correct types for function arguments in `splrep` and `splev`

Fix types for `x` and `y` arguments in `splrep` and `splev` functions to
use slices instead of `Vec`. This ensures compatibility with the
existing codebase. (master)

spartajet commented:
@hochej OK？
